parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/HadoopInputFile.java

```diff
@@ -66,7 +68,13 @@ public long getLength() {
 
   @Override
   public SeekableInputStream newStream() throws IOException {
-    return HadoopStreams.wrap(fs.open(stat.getPath()));
+    // specify random access always
+    return HadoopStreams.wrap(
+      awaitFuture(fs.openFile(stat.getPath())
```

Comment:
hmm won't this cause issue if the actual hadoop version is below 3.3? I think we may need to check the version through `VersionInfo.getVersion`.

Reply:
like I said, WiP 😄 

the shim library I'm doing does retrofit this to older versions, but I'm doing this to see what opportunities there are with basic adoption of 3.3.x apis. (https://github.com/steveloughran/fs-api-shim)

Mukund intends to add vectored stream reads to that shim lib too falling back to the positioned read operations, for versions <= 3.3.4

Reply:
Ah I see. Looking forward to that!

Reply:
people should still be upgrading all their project dependencies though. 3.3.2 should be the minimum.